app/views/exercises/show.html.erb

```diff
@@ -4,6 +4,22 @@
 <ul>
   <li><%= link_to 'Download project template (zip)', exercise_zip_url(@exercise) %></li>
 
+  <% unless @exercise.available_points.empty? %>
+    <%
+      available_points = @exercise.available_points.to_a.sort.map(&:name)
+      awarded_points = available_points & @exercise.points_for(current_user)
+      nonawarded_points = available_points - awarded_points
+      all_points = awarded_points.map {|pt| [pt, 'success']} + nonawarded_points.map {|pt| [pt, 'failure']}
+      all_points = all_points.sort {|pt1, pt2| Natcmp.natcmp(pt1[0], pt2[0]) }
+    %>
+    <li>
+      Got <%= awarded_points.size %> out of <%= pluralize(available_points.size, 'point') %>
+      (<%= raw(all_points.map {|pt| '<span class="' + pt[1] + '">' + h(pt[0]) + '</span>' }.join(', ')) %>).
```

Comment:
I have a weak preference for a formulation e.g. like this: "You have N out of M points for this exercise (...)." just because it makes a bit more sense when the exercise has not yet been attempted.